quantum/process_keycode/process_unicode_common.c

```diff
@@ -207,9 +207,10 @@ void register_hex(uint16_t hex) {
 }
 
 void register_hex32(uint32_t hex) {
-    bool onzerostart = true;
+    bool onzerostart  = true;
+    int  onzerodigits = ((unicode_config.input_mode == UC_WINC) ? 4 : 3);
     for (int i = 7; i >= 0; i--) {
-        if (i <= 3) {
+        if (i <= onzerodigits) {
```

Comment:
The underlying problem is that the default set of compose sequences used by WinCompose includes `<Compose> <u> <a>` and `<Compose> <U> <E>` (in my testing with WinCompose 0.9.11 there was no `<Compose> <u> <e>` sequence there, but the most recent version of `libx11` has that sequence too; however, the version with both capital letters would be matched if Caps Lock happens to be on). WinCompose accepts the matching compose sequence immediately instead of waiting for a possible continuation of the Unicode input sequence (this is probably not even a bug, because the normal compose sequences don't have any termination character and are intended to be matched immediately after the whole sequence is entered), therefore entering the Unicode input sequence is not possible if the hex number for the Unicode codepoint starts with `a` or `e`.

The change fixes the problem for `U+Axxx` and `U+Exxx` by always sending 5 hex digits to WinCompose (so that the first digit will be 0, and apparently there are no predefined compose sequences which start with `u0`). There will be a slight performance penalty due to always sending an extra key; maybe it does not really matter though.

However, the fix is not 100% complete, because the same problem would be triggered when sending `U+Axxxx` or `U+Exxxx` codes. But currently the `U+Axxxx` range has no defined codepoints, and the `U+Exxxx` range contains only “tag characters” (which are probably not useful as keyboard input) and “ideographic-specific variation selectors” (these potentially can be used in combination with actual ideographic characters, but I'm not sure about the actual support for this feature in any fonts or software).

Maybe a better fix would be to prepend `0` if the first non-zero hex digit to be sent is not a decimal digit (this would convert the sequences for `U+Axxxx`…`U+Fxxxx` to 6-digit hex numbers, but WinCompose should support that, and getting an unsupported 7-digit hex number won't be possible, because the maximum allowed value for an Unicode codepoint is `0x10FFFF`).

Reply:
I've reworked as you suggest, and in the process, cleaned up the implementation to make it more efficient.
The end result is that the new workaround only costs an additional 6 bytes of code.